Manifest.toml

```diff
@@ -0,0 +1,256 @@
+# This file is machine-generated - editing it directly is not advised
```

Comment:
Remove this whole file?

Reply:
Do you not want to include a `Manifest.toml`?  It can be helpful to track down why, for instance, improper upper bounds caused a test to fail in the future.

Reply:
I'm a big believer in not including `Manifest.toml`s in packages :) In my mind they record things that shouldn't be recorded in a package.

Not sure I understand the upper bound scenario.

Reply:
> I'm a big believer in not including Manifest.tomls in packages :)

It's your package, so by all means, manage it how you like!  I'll remove it.

> Not sure I understand the upper bound scenario.

The `Manifest.toml` records a "known good point" within the entire space of packages that the resolver can choose.  If you use `Pkg.add("Cairo")`, you do not necessarily get a working setup for `Cairo.jl`, since the resolver is free to choose among a potentially very wide number of packages.  If the dependency upper bounds are chosen perfectly, `Pkg.add()` should always work.  However, it's pretty common that somebody updates a package somewhere in your deps tree that breaks something higher up in the deps tree, eventually causing `Cairo.jl` to break.  This doesn't even have to be `Cairo.jl`'s fault; it could be a transitive dependency that causes the problem.

Bundling a `Manifest.toml` provides two tools to help out:
* If you are in the "just work, darnit" frame of mind, you can instantiate the `Manifest.toml` to get the versions that worked in the past (and thereby should always work, especially in light of JLL packages which extend instantiation's effect to binary packages).
* If you are trying to debug what versions of a dependency _should_ work, the `Manifest` at least gives you a known good point; you don't need to go and look at the Registry to see what versions would have been installed on your machine six months ago when you last modified this package; all that information is stored here.

Manifest.toml files bundled with packages aren't used for resolution or loading in any way during normal usage; you have to explicitly start Julia with `--project=` pointing at `Cairo.jl` for them to be used, and they'll only install their versions if you explicitly `instantiate`.

Reply:
> It's your package, so by all means, manage it how you like! I'll remove it.

Actually, it is not :) Not sure who is in charge, probably @lobingera?

Thanks for the explanation of the benefits, interesting. I think one main reason why I don't like to have `Manifest.toml`s in my packages is that VS Code will auto-activate an env if there is a `Manifest.toml` in a package folder, and that is something I don't want, so it throws a wrench into my normal workflow. Also, it just seems that almost all packages don't have it in the root, and I'm a fan of just following common practices with this kind of choice :)

Reply:
Ah, I had no idea VS Code did that.  Interesting.  I can see the reason it does it, but yes that would be frustrating.  This is where the concept of an "application" versus a "package" start to conflict a little bit.  :P

Reply:
From the VS Code extension point of view, it would be _great_ if a package project was called `Package.toml`, and an app project was called `Project.toml`, even if they have exactly the same syntax/content. Or some other way to distinguish a package from an app. It would just be really convenient for tools to be able to look at a folder and just say "ah, this is a package", or "ah, this is an app"...